### ISSUE TITLE
UW/SI2: computed diffusion data

### ISSUE THREAD

**Issue by tschaume** [materialsproject/MPContribsUsers#4]
In collaboration with @henry-wu, a workflow is developed to contribute computed diffusivities of e.g. Au in Cu, and link the data to the perfect solvent structure in MP.

_Pre-processing_:
- The abstract class `AbstractVaspDirCollParser` parses a collection of related (nested) VASP directories. It uses pymatgen’s powerful borg module to assimilate in parallel the available VASP in-/output data in the submitted root directory (e.g. “uw_diffusion” containing Cu{Cu,Au,..} folders). The constructor converts each directory with either a `vasprun.xml`, a `POSCAR`, or a `POSCAR` and an `OSZICAR` into a `ComputedEntry` or `ComputedStructureEntry` depending on whether the full structural information is available from a `CONTCAR` or `POSCAR`. The user can subsequently retrieve a specific entry via the `find_entry_for_directory` convenience function by providing a regex matching the targeted VASP subdirectory.
- The idea is then for the user to implement a new parser by inheriting from `AbstractVaspDirCollParser` and define the `compile` method. This method is designed and maintained by the user to extract all derived data like diffusion coefficients from the VASP directories and compile it into a `MPFile`.
- The UW/SI2-specific `VaspDirCollParser` extracts energy barriers, attempt frequencies and diffusion coefficients. It first uses the new MP REST endpoint `find_structure` via the according MPRester function to match the solvent to an existing structure in the MP and retrieve the corresponding `mp_id`. After extracting energy barriers and attempt frequencies, it generates T-dependent diffusion coefficients for each combination of solvent and solute based on MAST’s `DiffusionCoefficient.py`.

During the _post-processing_, the contributed data as well as interactive graphs are pulled from MP and incorporated into a possibly purely dynamic user web-app.

List of ToDo's:
- [x] develop `AbstractVaspDirCollParser` based on pymatgen's borg module,  
  see [mpcontribs.io.vaspdir](https://github.com/materialsproject/MPContribs/blob/d744d2d/mpcontribs/io/vaspdir.py) and _pre-processing_ above.
- [x] implement `DYNMAT` parsing, see pull request materialsproject/pymatgen#195
- [x] write UW/SI2 specific `VaspDirCollParser`,  
  see [mpcontribsusers.uw_si2.uw_si2](https://github.com/materialsproject/MPContribsUsers/blob/8ac0c2e/mpcontribsusers/uw_si2/uw_si2.py) and _pre-processing_ above.
- [ ] review the `DYNMAT` I/O object and its `get_phonon_frequencies` method  added in materialsproject/pymatgen#195 @henry-wu
- [ ] add missing `OUTCAR` files to test directory such that detour through the `DYNMAT` file  to obtain the phonon frequencies can be avoided @henry-wu
- [ ] review `uw_si2` module and update pull request #2 @henry-wu
- [ ] which meta-data would be interesting to collect during data aggregation? @henry-wu
- [x] develop rudimentary `MPFile` viewer, see materialsproject/MPContribs#8
- [x] improve book-keeping, see materialsproject/MPContribs#3
- [ ] release framework for beta-testers, see materialsproject/MPContribs#4
- [ ] iterate submission of output `MPFile` to test presentation of contributed data on front-end
- [ ] prepare end-to-end demo video
- [ ] start pull request to work on post-analysis of contributed data

**Comment by tschaume** [comment]
This issue has been superseded by the implementation of data ingestion from the published Excel sheet on FigShare. See f910db33279e44ab3efeb811f2b421a6547bca44
